main.js

```diff
@@ -1,6 +1,180 @@
+Vue.component("product-details", {
+  props: {
+    details: {
+      type: String,
+      required: true,
+      default: "books",
+    },
+  },
+  template: `<div class="product-details">
+              {{details}}
+              <button v-on:click="emitEventClick"> Emit Click !</button>
+              <div>
+        <p v-if="!reviews.length">There are no reviews yet.</p>
+        <ul v-else>
+            <li v-for="(review, index) in reviews" :key="index">
+              <p>Name: {{ review.name }}</p>
+              <p>Rating: {{ review.rating }}</p>
+              <p>Review: {{ review.review }}</p>
+            </li>
+        </ul>
+    </div>
+   
+   <product-review @review-submitted="addReview"></product-review>
+            </div>
+            
+  `,
+  data() {
+    return {
+      selectedBookId: 0,
+      reviews: [],
+    };
+  },
+  methods: {
+    emitEventClick: function () {
+      this.$emit("emit-event", this.selectedBookId);
+    },
+    addReview(productReview) {
+      this.reviews.push(productReview);
+    },
+  },
+});
+
+Vue.component("product-review", {
+
+  // form + v-model
+  template: `
+    <form class="review-form" @submit.prevent="onSubmit">
+    
+      <p class="error" v-if="errors.length">
+        <b>Please correct the following error(s):</b>
+        <ul>
+          <li v-for="error in errors">{{ error }}</li>
+        </ul>
+      </p>
+
+      <p>
+        <label for="name">Name:</label>
+        <input id="name" v-model="name">
+      </p>
+      
+      <p>
+        <label for="review">Review:</label>      
+        <textarea id="review" v-model="review"></textarea>
+      </p>
+      
+      <p>
+        <label for="rating">Rating:</label>
+        <select id="rating" v-model.number="rating">
+          <option>5</option>
+          <option>4</option>
+          <option>3</option>
+          <option>2</option>
+          <option>1</option>
+        </select>
+      </p>
+      <label> Would you recommend this product ? </label>
+      <p>
+      <input type="radio" v-model="radio" value="yes" name="yes_no">Yes</input>
+      </p>
+      <p>
+      <input type="radio" v-model="radio" value="no" name="yes_no">No</input>
+      </p>
+      
+      <p>
+        <input type="submit" value="Submit">  

```

Comment:
Use button instead input element
```suggestion
        <button type="submit" >Submit</button>  

```